platform-sdk/swirlds-merkledb/src/testFixtures/java/com/swirlds/merkledb/test/fixtures/TestType.java

```diff
@@ -153,18 +150,16 @@ public MerkleDbDataSource createDataSource(
                 final boolean enableMerging,
                 boolean preferDiskBasedIndexes)
                 throws IOException {
-            final MerkleDb database = MerkleDb.getInstance(dbPath, CONFIGURATION);
-            final MerkleDbTableConfig tableConfig =
-                    new MerkleDbTableConfig((short) 1, DigestType.SHA_384, size, hashesRamToDiskThreshold);
-            MerkleDbDataSource dataSource = database.createDataSource(name, tableConfig, enableMerging);
+            MerkleDbDataSource dataSource = new MerkleDbDataSource(
+                    dbPath, CONFIGURATION, name, size, hashesRamToDiskThreshold, enableMerging, preferDiskBasedIndexes);
             dataSource.registerMetrics(getMetrics());
             return dataSource;
         }
 
         public MerkleDbDataSource getDataSource(final Path dbPath, final String name, final boolean enableMerging)
                 throws IOException {
-            final MerkleDb database = MerkleDb.getInstance(dbPath, CONFIGURATION);
-            return database.getDataSource(name, enableMerging);
+            // Assuming initial capacity and hashes RAM/disk threshold are loaded from table metadata
+            return new MerkleDbDataSource(dbPath, CONFIGURATION, name, 0, 0, enableMerging, false);
```

Comment:
What do you think about creating a new constructor in MerkleDbDataSource and documenting it there?

Reply:
Done